BrewMesModulesParent/batch/src/main/java/com/brewmes/batch/BatchGetterImpl.java

```diff
@@ -17,4 +21,21 @@
     public Page<Batch> getBatches(int page, int size) {
         return batchRepository.findAll(PageRequest.of(page, size));
     }
+
+    public JsonObject getStaticBatchVariables(String id) {
+        List<Batch> batchList = batchRepository.findAll();
+        Collections.reverse(batchList);
+        for (Batch batch : batchList) {
+            if (batch.getConnectionID().equals(id)) {
+                JsonObject batchObject = new JsonObject();
+                batchObject.add("id", new JsonPrimitive(batch.getID()));
+                batchObject.add("type", new JsonPrimitive(batch.getProductType().productName));
+                batchObject.add("amount", new JsonPrimitive(batch.getAmountToProduce()));
+                batchObject.add("speed", new JsonPrimitive(batch.getDesiredSpeed()));
+
+                return batchObject;
+            }
+        }
+        return null;
+    }
```

Comment:
Then we have to make a class that implements the repository which implements the method and call that instead. 
On that implementation we need this exact code, we could but it wont save us any lines unless it was to be called multiple times.

Reply:
Nope, you add -> List<Batch> getBatchFromConnectionID(String connectionID); To the repository interface. This works because the batch entity has a similar field. Spring and mongoDB will take care of the rest.